libbeat/logp/file_rotator.go

```diff
@@ -73,16 +75,26 @@ func (rotator *FileRotator) WriteLine(line []byte) error {
 	}
 
 	line = append(line, '\n')
+
+	rotator.currentLock.RLock()
 	_, err := rotator.current.Write(line)
```

Comment:
I'd say having `currentSize` slightly out of date is fine, we just use it for the rotator so it's ok if that starts a bit late.